Tasks/Gradle/CodeAnalysis/Common/CodeAnalysisResultPublisher.ts

```diff
@@ -41,7 +41,9 @@ export class CodeAnalysisResultPublisher {
 
             for (var resultFile of analysisResult.resultFiles) {
                 let extension = path.extname(resultFile);
-                let artifactName = `${prefix}_${analysisResult.moduleName}_${analysisResult.toolName}${extension}`;
+                let reportName = path.basename(resultFile, extension);
+
+                let artifactName = `${prefix}_${reportName}_${analysisResult.toolName}${extension}`;
```

Comment:
I found a bug with the old implementation - a module can have multiple reports (e.g. main.xml and test.xml), but the old renaming logic did not distinguish between them - some reports were not uploaded! I fixed this by including the report name and added tests.